src/api/auth/login.rs

```diff
@@ -22,20 +23,26 @@ impl Instance {
         // We do not have a user yet, and the UserRateLimits will not be affected by a login
         // request (since login is an instance wide limit), which is why we are just cloning the
         // instances' limits to pass them on as user_rate_limits later.
-        let mut shell = UserMeta::shell(Rc::new(RefCell::new(self.clone())), "None".to_string());
+        let mut shell =
+            UserMeta::shell(Rc::new(RefCell::new(self.clone())), "None".to_string()).await;
         let login_result = chorus_request
             .deserialize_response::<LoginResult>(&mut shell)
             .await?;
         let object = self.get_user(login_result.token.clone(), None).await?;
         if self.limits_information.is_some() {
             self.limits_information.as_mut().unwrap().ratelimits = shell.limits.clone().unwrap();
         }
+        let mut identify = GatewayIdentifyPayload::common();
+        let gateway = Gateway::new(self.urls.wss.clone()).await.unwrap();
```

Comment:
Because if we don't include it, then we'd have to put Gateway as optional :C idk, would that be okay? /gen

Reply:
As mentioned on Discord: I might modify the ratelimiter to take a trait Impl instead of UserMeta. That way, the mocked user doesn't have to have a Gateway attached to it.

Reply:
I'll merge and make this an issue, because this seems to be a little more involved than I'd like for now